test/test_utils.py

```diff
@@ -225,3 +225,34 @@ def test_invalid_png_data(self):
                           imageinfo.identify, data)
         self.assertRaises(imageinfo.UnrecognizedFormat,
                           imageinfo.identify, data)
+
+
+class TestEncodeFilename(unittest.TestCase):
+
+    @unittest.skipIf(os.path.supports_unicode_filenames , "unicode is supported")
+    def test_EncodeFilenames_Nosupport(self):
+        self.assertEqual(b'\xe5\xb0\x8f\xe5\xae\xa4', util.encode_filename("小室"))
+        self.assertEqual(b'\xe5\xb0\x8f', util.encode_filename('小'))
+
+    @unittest.skipUnless(os.path.supports_unicode_filenames and (sys.platform == 'darwin'), "requires unicode support and darwin os")
+    def test_EncodeFilenames_support_darwin(self):
+        self.assertEqual(b'\xe5\xb0\x8f\xe5\xae\xa4', util.encode_filename("小室"))
+
+    @unittest.skipUnless(os.path.supports_unicode_filenames and (sys.platform != 'darwin'), "requires unicode support and other than darwin os")
+    def test_EncodeFilenames_support(self):
+        self.assertEqual("小室", util.encode_filename("小室"))
+
+@unittest.skipIf(sys.getdefaultencoding()=='utf-8' , "requires utf-8 encoding")
```

Comment:
The test is skipped if `sys.getdefaultencoding() == 'utf-8'`, but the message says `requires utf-8 encoding`, it looks incorrect, the test should actually be run if the encoding is 'utf-8'

On my linux system:

```bash
python3 -c 'import sys; print(sys.getdefaultencoding())'
utf-8
```

But:

```
test_AnySystem (test.test_utils.TestDecodeFilename) ... skipped 'requires utf-8 encoding'
test_WithUnicodeSupport (test.test_utils.TestDecodeFilename) ... skipped 'requires utf-8 encoding'
test_WithoutUnicodeScript (test.test_utils.TestDecodeFilename) ... skipped 'requires utf-8 encoding'
test_EncodeFilenames_Nosupport (test.test_utils.TestEncodeFilename) ... ok
test_EncodeFilenames_support (test.test_utils.TestEncodeFilename) ... skipped 'requires unicode support and other than darwin os'
test_EncodeFilenames_support_darwin (test.test_utils.TestEncodeFilename) ... skipped 'requires unicode support and darwin os'
test_cyrillic (test.test_utils.TranslateArtistTest) ... ok
```

Reply:
okay i got the typo, but
@zas yes, on my machine also i have 'utf-8' , and tests indeed run right if its `utf-8` but appveyor compares it with weird symbols and fails the tests :/ 
see the last WIP push, it should have skipped the test of its not utf-8 . I cant figure out whats going on. any help please?